source/common/network/connection_impl.cc

```diff
@@ -44,6 +44,16 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
 
 std::atomic<uint64_t> ConnectionImpl::next_global_id_;
 
+std::unique_ptr<ConnectionImpl>
+ConnectionImpl::createServerConnection(Event::Dispatcher& dispatcher, ConnectionSocketPtr&& socket,
+                                       TransportSocketPtr&& transport_socket) {
+  // make_shared<> requires a public constructor so we can't use it here.
+  std::unique_ptr<Network::ConnectionImpl> conn(new Network::ConnectionImpl(
+      dispatcher, std::move(socket), std::move(transport_socket), true));
+  conn->transport_socket_->setTransportSocketCallbacks(*conn);
```

Comment:
Can you add a comment here and below (or just refer one to the other) of why we need to do this for future readers?

Reply:
Will do.